vpr/src/pack/constraints_report.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * constraints_report.cpp
+ *
+ *  Created on: Dec. 7, 2021
+ *      Author: khalid88
+ */
+
+#include "constraints_report.h"
+
+struct region_content_info {
+    Region reg;
+    std::vector<int> block_type_counts;
+};
+
+void check_constraints_filling() {
+    //GridTileLookup class provides info needed for calculating number of tiles covered by a region
+    //TO-DO: Find better way of accessing grid tiles so you don't have to initialize it in two different locations (initial placement and here)
+    GridTileLookup grid_tiles;
```

Comment:
it's ok to have it initialize in two places, just have the constructor call the initialization routine